server/routes/ratings.js

```diff
@@ -13,71 +14,60 @@ router.get('/', function (req, res) {
+            return models.Snippet.findById(snippetId, {transaction: t}).then( function (s) {
+              return s.update({avg: avg}, {transaction: t}).then( function () {
+                return models.User.findById(s.UserId, {transaction: t}).then( function (user) {
+                  return models.Snippet.aggregate('avg', 'avg', { where : { UserId : user.id }, dataType: 'float', transaction: t }).then(function (totalAvg){
+                    return user.update({avg: totalAvg}, {transaction: t}).then(function () {
+                      return new Promise(function (resolve) {
+                        resolve({rating: rating.toJson(), avg: avg.toFixed(2)});
+                      });
+                    });
+                  });
+                });
+              });
+            });
           });
         });
-      }).catch(function () {
-        res.status(422).send('error');
-      });
-    } else {
-      var attributes = {
-        SnippetId: snippetId,
-        value: req.body.value,
-        UserId: userId
-      };
-      var new_rating = models.Rating.build(attributes);
-      new_rating.save({ validate: false, logging: true}).then(function (new_rating) {
-        models.Rating.sum('value', { where : { SnippetId: snippetId } }).then( function (sum) {
-          sumRatings = sum;
-          models.Rating.count({ where : { SnippetId: snippetId } }).then( function (c){
-            var average = 0;
-            if(c>0){
-              average = (sumRatings/c);
-            }
-            res.status(200).send({rating: new_rating, avg: average.toFixed(2)});
+      } else {
+        return models.Rating.create(attributes, {transaction: t}).then( function (newRating) {
+          return models.Rating.aggregate('value', 'avg', { where : { SnippetId : snippetId }, dataType: 'float', transaction: t }).then( function (avg) {
+            return models.Snippet.findById(snippetId, {transaction: t}).then( function (s) {
+              return s.update({avg: avg}, {transaction: t}).then( function () {
+                return models.User.findById(s.UserId, {transaction: t}).then( function (user) {
+                  return models.Snippet.aggregate('avg', 'avg', { where : { UserId : user.id }, dataType: 'float', transaction: t }).then( function (totalAvg) {
+                    return user.update({avg: totalAvg}, {transaction: t}).then(function () {
+                      return new Promise(function (resolve) {
+                        resolve({rating: newRating.toJson(), avg: avg.toFixed(2)});
+                      });
+                    });
+                  });
+                });
+              });
+            });
           });
         });
-      }).catch(function () {
-        res.status(422).send('error');
-      });
-    }
+      }
+    });
+  }).then( function (data) {
+    res.status(200).send(data);
```

Comment:
I think it's not a problem ... Generally IMO HTTP status codes are broken so lets just be consistent and check how was that done in previous actions.